File: escenario1/src/escenario1/VehicleAbstractFactory.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package escenario1;

/**
 *
 * @author Dell
 */
abstract class VehicleAbstractFactory {
    abstract Vehicle createVehicle(VehicleType vehicletype, int cylinderCapacity, int modelYear, double valuation);
}

File: src/Car/VehicleAbstractFactory.ts
import { Vehicle } from './Vehicle';
import { VehicleType } from './VehicleType';

export abstract class VehicleAbstractFactory {
  abstract createVehicle(
    vehicletype: VehicleType,
    cylinderCapacity: number,
    modelYear: number,
    valuation: number,
  ): Vehicle;
}
